tools/lsvast/lsvast.cpp

```diff
@@ -141,11 +141,17 @@ caf::expected<Kind> classify(const std::filesystem::path& path) {
   const auto is_directory = std::filesystem::is_directory(path, err);
   if (err)
     return caf::make_error(vast::ec::filesystem_error,
-                           "Invalid path: ", err.message());
-
-  if (is_directory && path.stem() == "vast.db")
-    return Kind::DatabaseDir;
-
+                           "Invalid path: " + path.string(), err.message());
+  if (is_directory) {
+    auto version_file = path / "VERSION";
+    auto is_regular_file = std::filesystem::is_regular_file(version_file, err);
+    if (err)
+      return caf::make_error(vast::ec::filesystem_error,
+                             "Invalid path: " + version_file.string(),
```

Comment:
Do we need the `.string()`? I thought you fixed the `fmt` specializations for `std::filesystem::path`

Reply:
This isn't going through {fmt} yet.

Reply:
For what it's worth, we do plan to do this in `vast/error.hpp` eventually:

```cpp
namespace vast {

/// Create an error with the error context formatted via `fmt::format`.
/// @param code The VAST error code.
/// @param args The arguments to forward to `fmt::format` to construct the error
/// context.
template <class... Args>
auto make_error(ec code, Args&&... args) {
  if constexpr (sizeof...(args) <= 1)
    return caf::make_error(code, std::forward<Args>(args)...);
  else
    return caf::make_error(code, fmt::format(std::forward<Args>(args)...));
}

} // namespace vast

namespace caf {

/// Poison `caf::make_error` for `vast::ec` to ensure users go through the
/// properly formatted version `vast::make_error`.
template <class... Args>
error make_error(::vast::ec, Args&&...) = delete;

} // namespace caf
```